app/queries/servers/channel.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Database, Q} from '@nozbe/watermelondb';
+
+import {MM_TABLES} from '@constants/database';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type ChannelInfoModel from '@typings/database/models/servers/channel_info';
+
+const {SERVER: {CHANNEL}} = MM_TABLES;
+
+export const prepareMyChannelsForTeam = async (operator: ServerDataOperator, teamId: string, channels: Channel[], channelMembers: ChannelMembership[]) => {
+    const allChannelsForTeam = await queryAllChannelsForTeam(operator.database, teamId);
+    const channelInfos: ChannelInfo[] = [];
+    const memberships = channelMembers.map((cm) => ({...cm, id: cm.channel_id}));
+
+    for await (const c of channels) {
+        const storedChannel = allChannelsForTeam.find((sc) => sc.id === c.id);
+        let storedInfo: ChannelInfoModel;
+        let member_count = 0;
+        let guest_count = 0;
+        let pinned_post_count = 0;
+        if (storedChannel) {
+            storedInfo = (await storedChannel.info.fetch())[0] as ChannelInfoModel;
```

Comment:
I know we have the 1:1 relation issue with WDB and that's why were are accessing index 0.  But other devs might not know this at first sight.

Maybe in the future, as an improving, we can write methods within Model that returns just what we need - like [Readers](https://nozbe.github.io/WatermelonDB/Writers.html#advanced-readers)

Reply:
Interesting.. I'll give this a try

Reply:
Reader was added on 0.23, this requires a dependency update... will go ahead and try it

Reply:
definitely will be trying this on a different PR, `@writer` and `@reader` are in fact replacing the deprecated `@action` in fact `database.action(() => {}` has been deprecated as well. We will need to deal with all this.

Reply:
So yes, we need to do this BUT I've also found a way to generate a 1:1 relation that works both ways. I guess a lot of changes are coming to the database layer (models, schema, operators, etc..)

Reply:
Added some db model changes to this PR, please re-review